src/test/scala/chiselTests/reflect/DataMirrorSpec.scala

```diff
@@ -122,4 +123,23 @@ class DataMirrorSpec extends ChiselFlatSpec {
     ChiselStage.emitCHIRRTL(new MyModule)
   }
 
+  "isFullyAligned" should "work" in {
+    class InputOutputTest extends Bundle {
+      val incoming = Input(DecoupledIO(UInt(8.W)))
+      val outgoing = Output(DecoupledIO(UInt(8.W)))
+    }
+    assert(!DataMirror.isFullyAligned(new InputOutputTest()))
+
+    assert(DataMirror.isFullyAligned(new InputOutputTest().incoming))
+    assert(DataMirror.isFullyAligned(new InputOutputTest().outgoing))
+    assert(DataMirror.isFullyAligned(Input(new InputOutputTest())))
+    assert(DataMirror.isFullyAligned(Input(new InputOutputTest()).outgoing))
+    assert(DataMirror.isFullyAligned(Output(new InputOutputTest())))
+    assert(DataMirror.isFullyAligned(Output(new InputOutputTest()).incoming))
+    assert(DataMirror.isFullyAligned(Output(new InputOutputTest()).incoming.ready))
+
+    assert(!DataMirror.isFullyAligned(new DecoupledIO(UInt(8.W))))
```

Comment:
might be nice to see a few more negative tests...? can you add a not-coerced DecoupledIO into the InputOutputTest bundle and show that it is also not fully aligned?

Reply:
Absolutely, thanks for the suggestion!

Reply:
Added some tests, thanks!